[Vectorize] Vectorize metadata filtering edits (#12249)
cloudflare/cloudflare-docs: content/vectorize/learning/metadata-filtering.md

Optional `filter` property on `query()` method specifies metadata filter:

| Operator | Description |
| -------- | ----------- |
| `$eq`    | Equals      |

> | `$ne`    | Not equeals |

Reviewer: ```suggestion
| `$ne`    | Not equals |
```

- `filter` must be non-empty object whose compact JSON representation must be less than 2048 bytes.
- `filter` object keys cannot be empty, contain `" | .` (dot is reserved for nesting), start with `$`, or be longer than 512 characters.
- `filter` object non-nested values can be `string`, `number`, `boolean`, or `null` values.

> ### Namespace vs. metadata filtering

Author: @netgusto @ndisidore anything else to callout here?

Would saying namespace=logical partition vs metadata=arbitrary tags help?

> ### Namespace vs. metadata filtering

Reviewer: ```suggestion
### Namespace versus metadata filtering
```

> Both [namespaces](/vectorize/learning/insert-vectors/#namespaces) and metadata filtering narrow the vector search space for a query.  When evaluating both filter types, considerations include:

Reviewer: ```suggestion
Both [namespaces](/vectorize/learning/insert-vectors/#namespaces) and metadata filtering narrow the vector search space for a query.  Consider the following when evaluating both filter types:
```

> - Namespace filter is applied first then metadata filter(s). 

Reviewer: ```suggestion
- A namespace filter is applied before metadata filter(s). 
```

> - A vector can only be part of a single namespace with the documented [limits](/vectorize/platform/limits/). Whereas, vector metadata can contain multiple key-value pairs up to [metadata per vector limits](/vectorize/platform/limits/). Metadata values support different types (`string`, `boolean`, etc) so offer more flexibility.

Reviewer: ```suggestion
- A vector can only be part of a single namespace with the documented [limits](/vectorize/platform/limits/). Vector metadata can contain multiple key-value pairs up to [metadata per vector limits](/vectorize/platform/limits/). Metadata values support different types (`string`, `boolean`, and others), therefore offering more flexibility.
```
